### DISCUSSION TITLE
Odd DNS error

### DISCUSSION THREAD

**Discussion by x-FBi** [angristan/wireguard-install#240] [Help] [answered]
After running the script my box loses the ability to resolve DNS. (apt doesn't resolve and I cannot ping google.com)
I need to manually correct the issue.  I'm a newb, so this was the fix I came up with.

```
# The primary network interface
auto eth0
iface eth0 inet static
address <eth0 IP>
netmask 255.255.255.128
dns-nameserver 1.1.1.1 #Your DNS here
dns-nameserver 1.0.0.1 #Your secondary DNS here
gateway <VPS GATEWAY IP>
```

Again, not sure why, but it's only happened once out of 5 servers, and it consistently does it on this ONE server.

**Comment by angristan** [comment]
What DNS resolvers did you choose during the script execution?

**Reply by x-FBi** [reply]
Same DNS for all servers,
1.1.1.1
1.0.0.1

Thanks for the reply.

**Reply by angristan** [reply]
Ok. I'm not sure what your fix consists of, what exactly did you change in your network configuration?

**Reply by x-FBi** [reply]
In the discussion,
I had to edit my
/etc/network/interface

And add the two DNS-nameserver lines.

**Reply by x-FBi** [reply]
Debian 10.

**Reply by angristan** [reply]
Ok, if you are able to reproduce the issue again, can you run the `dig` command and see what happens? There should be a "server" section. Your issue indicates that the DNS config might not be applied correctly

![screenshot_2021-06-05_10-49-07](https://user-images.githubusercontent.com/11699655/120885930-c8b19600-c5eb-11eb-8cf5-ef61dbffe197.png)

**Reply by x-FBi** [reply]
dig didn't work -- command not found. (yes, I get why it wouldn't be found, but after I fix nameserver to get DIG, it would be working)

**Comment by x-FBi** [comment, marked as answer]
Your script installs resolvconf

> apt-get install -y wireguard iptables resolvconf qrencode

If nothing is setup on the server, no DNS is set.
The user must now edit /etc/network/interfaces

Read Below:


`The resolvconf program

The resolvconf program keeps track of system information about the currently available nameservers. It should not be confused with the configuration file resolv.conf, which unfortunately has a nearly identical name. The resolvconf program is optional on a Debian system.

The configuration file resolv.conf contains information about the nameservers to be used by the system. However, when multiple programs need to dynamically modify the resolv.conf configuration file they can step on each other and the file can become out-of-sync. The resolvconf program addresses this problem. It acts as an intermediary between programs that supply nameserver information (e.g. dhcp clients) and programs that use nameserver information (e.g. resolver).

When resolvconf is properly installed, the resolv.conf configuration file at /etc/resolv.conf is replaced by a symbolic link to /etc/resolvconf/run/resolv.conf and the resolver instead uses the configuration file that is dynamically generated by resolvconf at /etc/resolvconf/run/resolv.conf.

The resolvconf program is generally only necessary when a system has multiple programs that need to dynamically modify the nameserver information. In a simple system where the nameservers do not change often or are only changed by one program, the resolv.conf configuration file is adequate.

If the resolvconf program is installed, you should not edit the resolv.conf configuration file manually as it will be dynamically changed by programs in the system. If you need to manually define the nameservers (as with a static interface), add a line something like the following to the interfaces configuration file at /etc/network/interfaces:

dns-nameservers 12.34.56.78 12.34.56.79

Place the line indented within an iface stanza, e.g., right after the gateway line. Enter the IP addresses of the nameservers you need to use after dns-nameservers. Put all of them on one line separated by spaces. Don't forget the "s" on the end of dns-nameservers.

The resolvconf program is a fairly new addition to Debian and many older programs need to be updated or reconfigured to work properly with it. If you have problems, see /usr/share/doc/resolvconf/README. It has lots of information on making other programs get along with resolvconf. `
